python/mxnet/metric.py

```diff
@@ -81,7 +81,7 @@ def update(self, labels, preds):
         assert len(labels) == len(preds)
         for pred, label in zip(preds, labels):
             self.sum_metric += self._feval(label, pred)
-        self.num_inst += 1
+            self.num_inst += 1
```

Comment:
I find this fix should be write. It is equivalent to += len(label) if placed outside the loop. I have tried to evaluate it in a batch way, however the tests failed. I think we should keep compatible with it. @antinucleon @tqchen @piiswrong